docs/memorymanagement.md

```diff
@@ -0,0 +1,108 @@
+# C#/WinRT Object Lifetime and Reference Tracking
+
+## Overview
+
+C#/WinRT is a WinRT projection for C# which at a high level generates wrapper C# types to represent
+WinRT types. The lifetime of any of these instantiated C# types is managed by the .NET garbage collector
+as with any C# object. But as a WinRT projection, the lifetime of the WinRT objects it wraps is
+managed by COM reference tracking. The XAML runtime also manages the lifetime of XAML / WinUI objects
+and has its own reference tracking that interacts with .NET and its garbage collector. This document
+serves the purpose of documenting how C#/WinRT interacts with all 3 systems to correctly manage the
+lifetime of projected WinRT objects.
+
+### COM reference tracking

```

Comment:
It might be easier to augment this with a series of enumerated steps in the process.